### ISSUE TITLE
[CLOSED] Adding Masked Language Modelling

### ISSUE THREAD

**Issue by jeswan** [nyu-mll/jiant-v1-legacy#1030]
<a href="https://github.com/pruksmhc"><img src="https://avatars3.githubusercontent.com/u/10094008?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [pruksmhc](https://github.com/pruksmhc)**
_Sunday Mar 08, 2020 at 13:22 GMT_
_Originally opened as https://github.com/nyu-mll/jiant/pull/1030_

----

Adding Masked Language Modeling Task for RoBERTa and ALBERT
What this version of MLM supports: RoBERTa and ALBERT embedders.
Additionally, we fix the get pretrained_lm head function for Transformer-based embedders

**Performance Tests:**
We tested this on CCG and QQP, on both making sure that RoBERTa MLM + QQP -> target task and RoBERTa MLM + CCG -> target were getting reasonable numbers (meaning that MLM perplexity was decreasing and QQP/CCG performance should be close to RoBERTa without MLM training. 

| Task      | Task performance (best RoBERTa single finetuned on task)  | Task performance (multitask trained with MLM with RoBERTa)    | Perplexity after training
| ------------- |:-------------:| :-------------:| :-------------:| 
| CCG   | 0.96 |  0.953 | 2.90563|
| QQP     | 0.898     | 0.853 | 2.45982|

| Target task | Task Performance | CCG w/o MLM performance 
| ------------- |:-------------:| :-------------:|
WIC | 0.760 | 0.716| 
COPA | 0.860 | 0.55| 
CB | 0.820 |0.791 |
RTE | 0.834 | 0.794 | 
CSenseQA | 0.739 | |
BoolQ | 0.837 | 0.84 |
MultiRC | 0.655 | 0.42| 
ReCoRD | 0.82976 | 0.838|
Cosmos | 0.81 | 0.774|

----

_**[pruksmhc](https://github.com/pruksmhc)** included the following code: https://github.com/nyu-mll/jiant/pull/1030/commits_
